### PR TITLE
Bugfix/issue#3105 selector max id nested at statements

### PR DESCRIPTION
**Pull request by BenjaminWFox**
> Which issue, if any, is this issue related to?

#3105 

> Is there anything in the PR that needs further explanation?

I haven't done a lot of these (edits/PRs) so I'm not sure if I missed something, but I couldn't find a scenario in the tests where the the check for 'atrule' was useful. Everything seemed to work as I expected when I removed it.

### CODE REVIEW

File: lib/rules/selector-max-id/__tests__/index.js [hunk 1]
```diff
@@ -167,6 +167,26 @@ testRule(rule, {
...
       description: "media query: nested"
+    },
+    {
+      code: "#foo { @if ($p == 1) { #bar {} } }",
+      description: "@if statement: nested"
+    },
+    {
+      code: ".foo { @if ($p == 1) { #bar #baz {} } }",
+      description: "@if statement: nested compound selector"
+    },
```
Comment (hudochenkov) [inline lib/rules/selector-max-id/__tests__/index.js]: I think tests containing `@if` should go to `test` function with `syntax: "scss"`.

File: lib/rules/selector-max-id/__tests__/index.js [hunk 2]
```diff
@@ -198,6 +218,52 @@ testRule(rule, {
...
+      column: 1
+    },
+    {
+      code: ".foo { @if ($p == 1) { #bar #baz #foo {} } }",
+      description:
+        "nested compound selector: greater than max id selectors, nested @if statement",
+      message: messages.expected(".foo #bar #baz #foo", 2),
+      line: 1,
+      column: 24
+    },
```
Comment (hudochenkov) [inline lib/rules/selector-max-id/__tests__/index.js]: I think tests containing `@if` should go to `test` function with `syntax: "scss"`.

**Comment by Arcanemagus** [comment]
@BenjaminWFox You should either merge changes into this branch, or rebase the branch on the new changes. Cherry picking the commits from `master` into here just causes confusion in the log when this is merged unless all of these commits are squashed together.

**Comment by BenjaminWFox** [comment]
@Arcanemagus Hmm, I thought I rebased but I haven't dealt with upstream/forked repos before - I probably got confused trying to get the right updates from the right place.

Should I just squash them then? And would that be with something like `git rebase -i <after-specific-commit>`? Thanks.

**Comment by Arcanemagus** [comment]
Actually as @hudochenkov just pointed out to me in a different PR, _all_ PR's here are squashed so it's not a huge issue.

For your own reference though:

The simplest method is to make sure your local `master` matches the upstream `master`, then run `git rebase master`. That makes your commit(s) move to having the current `master` as their parent commit.

Alternatively you can also do `git merge master`, which creates a separate commit that merges the current state of `master` into your branch, leaving your existing commits untouched.

**Pull request merged**